### ISSUE TITLE
VictoriaLogs failing to parse rfc5424 message correctly via syslog

### ISSUE THREAD

**Issue by johnl** [VictoriaMetrics/VictoriaMetrics#7776]
### Describe the bug

I'm sending what I believe to be a valid rfc5424 message with structured data to VictoriaLogs via syslog and the resulting log is missing the message.

### To Reproduce

Send the following log line to VictoriaLogs via the tcp syslog receiver:

```
<134>1 2024-12-09T18:25:35.401631+00:00 ps999 account-server - - [sd@51059 project="secret" ] 1.2.3.4 - - [09/Dec/2024:18:25:35 +0000] "PUT someurl" 201 - "-" "-" "container-updater 1283500" 0.0010 "-" 1531 0
```

and the entry shows up with a missing `_msg` field (UI says "missing _msg field; see https://docs.victoriametrics.com/victorialogs/keyconcepts/#message-field"). The `_time`, `app_name`, `facility`, `hostname`, `msg_id`, `priority`, `proc_id` and `severity` are all present and look correct. `format` is reported as `rfc5424`. The structured data is not available for searching, so it looks like that has not been parsed properly.

### Version

victoria-logs-20241208-233242-tags-v1.3.2-victorialogs-0-g6fe079dbfb

**Comment by AndrewChubatiuk** [comment]
hey @johnl 
how do you send a message to victorialogs syslog address?
tried to set docker driver to syslog with `rfc5424` format, but cannot reproduce any issues

**Comment by johnl** [comment]
I'm sending the syslog messages to victorialogs using rsyslogd, with a config like this:

```
   action(
     type="omfwd"
     target="hostnamehere"
     protocol="tcp"
     template="RSYSLOG_SyslogProtocol23Format"
     action.resumeRetryCount="-1"
     queue.type="LinkedList"
     queue.filename="remote_server_hostnamehere"
     queue.spoolDirectory="/var/spool/rsyslog"
     queue.saveonshutdown="on"
     queue.maxdiskspace="500m"
   )
```

and running victorialogs with the arguments:

```
./victoria-logs-prod -syslog.listenAddr.tcp=:514
```


I notice that the rsyslog docs at https://www.rsyslog.com/doc/configuration/templates.html say:

> RSYSLOG_SyslogProtocol23Format - the format specified in IETF’s internet-draft ietf-syslog-protocol-23, which is very close to the actual syslog standard [RFC5424](https://tools.ietf.org/html/rfc5424) (we couldn’t update this template as things were in production for quite some time when RFC5424 was finally approved).

So it's possible the forward isn't strictly RFC5424, but I can't see how my lines are incompatible.

This issue in rsyslog says it's related to a trailing \n perhaps https://github.com/rsyslog/rsyslog/issues/4384 but some lines are ingested fine, some are not, so I'm not sure this is related.

**Comment by AndrewChubatiuk** [comment]
hey @johnl 
Can you try changes from PR #7801?

**Comment by johnl** [comment]
yes, those changes fix the problem! Thanks @AndrewChubatiuk

**Comment by valyala** [comment]
The bugfix for this issue has been included in [v1.4.0-victorialogs](https://docs.victoriametrics.com/victorialogs/changelog/). Closing the issue as fixed.
